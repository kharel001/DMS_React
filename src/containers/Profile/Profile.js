import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import { Typography } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
    root: {
        '& .MuiTextField-root': {
            margin: theme.spacing(0.5),
            width: '50ch',

        },
        display: 'flex',
        flexDirection: 'column',
        width: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        margin: theme.spacing(2),

    },
    labelItem: {
        margin: theme.spacing(1),
        color: 'black',
    },
    textFieldItem: {

    },
    componentItem: {
        display: 'flex',
        flexDirection: 'column',

    },
}));

/**
 * This shows the seditable profile page.
 * @returns {Profile} - Returns a list of input fields for user profile 
 * @argument {Profile} - No Arguments
 */

const Profile = () => {

    const classes = useStyles();

    const [state, setState] = React.useState([
        { title: "Full Name", value: "" },
        { title: "Hospital Name", value: "" },
        { title: "Province", value: "" },
        { title: "District", value: "" },
        { title: "Town/ City", value: "" },
    ])

    const handleChange = (event, index) => {
        // console.log(event.target.value);
        let element = state[index];
        element = {
            ...element,
            value: event.target.value
        }
        let newState = [...state];
        newState.splice(index, 1, element);
        setState(newState);
    }

    return (
        <div className={classes.root}>
            <div>
                <Typography variant="h5" >User Profile</Typography>
            </div>
            <form noValidate autoComplete="off">
                {state.map((form, i, array) => {
                    return (<div className={classes.componentItem}>
                        <div className={classes.labelItem}>
                            <Typography>{form.title}</Typography>
                        </div>
                        <div className={classes.textFieldItem}>
                            <TextField
                                id="outlined-multiline-flexible"
                                // label="Multiline"
                                multiline
                                rowsMax={8}
                                value={form.value}
                                onChange={(event) => handleChange(event, i)}
                                variant="outlined"
                                size="small"
                            />
                        </div>
                    </div>)
                })}

            </form>
        </div>
    )
}

export default Profile;