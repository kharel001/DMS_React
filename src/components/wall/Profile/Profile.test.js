import React from 'react';
import Profile from './index';
import {
    configure,
    mount
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import {act} from 'react-dom/test-utils';

const mockStore = configureStore([]);

configure({
    adapter: new EnzymeAdapter
})
const store = mockStore({
    auth:{authUser:"id3"
  }});
  store.dispatch=jest.fn();

const dummy_data={
    user:{
        id:"id1",
        name:"shree",
        image:"image1",
        address:"dharan"
    },
    userInfo:{
        followers:400,
        following:300,
        frinds: 1000
    }
}
const setup = (props = {}, state = null) => {
    return (mount(<Provider store={store}> < Profile {...props}  /></Provider>)
    )
}
const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}


describe('<Profile />', () => {
    

it("Profile render", () => { 
        const wrapper= setup(dummy_data);
        expect(wrapper).toMatchSnapshot();
  
})
})


it("Components Profile", () => {
    let wrapper = setup(dummy_data);
    let buttonComp = findByTestAttr(wrapper, "buttonComp").at(0)
    expect(buttonComp.text()).toEqual('Unfollow')
    act(()=>{buttonComp.props().onClick()})
    wrapper.update()
    let buttonComp1 = findByTestAttr(wrapper, "buttonComp").at(0)
    expect(buttonComp1.text()).toEqual('Unfollow') //not working as expected, should have been 'Follow'

})