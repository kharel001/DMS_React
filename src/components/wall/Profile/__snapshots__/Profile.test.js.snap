// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> Profile render 1`] = `ReactWrapper {}`;
